### ISSUE TITLE
Getting text attributes.

### ISSUE THREAD

**Issue by shade-of-noon** [neovim/neovim#13349]
Hello.

I'm wondering if there's a reason/difficulty wrt being able to get the highlight attributes of the text under the cursor or, more generally, anywhere in the editor's viewport.

I'm aware of functions like `synID` and friends, and `nvim_get_hl_by_id` and friends. Nonetheless these functions work on syntax ID's/groups and so, if I'm not mistaken, they can't tell me the attributes of text that's been marked by a matched search pattern (hlsearch) or by `matchadd`, nor do they allow me to get the attributes of text at an arbitrary coordinate of the editor's viewport (relevant in functional tests, maybe).

In short IIUC, currently there's no way of getting the attributes of text highlighted interactively or at an arbitrary coordinate. And so I ask:
- Are there reasons blocking Vim/Neovim from having this feature?
- Is there work being done in relation to this issue?

Excuse my ignorance (I tried searching and failed) and thanks for any answers.

**Comment by justinmk** [comment]
Nvim 0.9 has `:Inspect` and `vim.inspect_pos()`.
